activerecord/test/cases/base_test.rb

```diff
@@ -1412,6 +1412,12 @@ def test_default_values_are_deeply_dupped
     assert_not_includes SymbolIgnoredDeveloper.columns_hash.keys, "first_name"
   end
 
+  test "#columns_hash raises an error if the record has an empty table name" do
+    assert_raises(ActiveRecord::ActiveRecordError) do
+      FirstAbstractClass.columns_hash
```

Comment:
Can you also verify the error message? `ActiveRecordError` is the superclass for many other errors. So it'd safer to make sure we're getting the right error.